src/main/java/org/mongoflink/sink/MongoBulkWriter.java

```diff
@@ -49,30 +55,36 @@
 
     private final boolean flushOnCheckpoint;
 
+    private final boolean upsertEnable;
+    private final String[] upsertKeys;
+
     private final RetryPolicy retryPolicy = new RetryPolicy(3, 1000L);
 
-    private transient volatile boolean initialized = false;
```

Comment:
We should keep `initialized ` to delay the schedule of flushes.

Reply:
done